### PR TITLE
backend/enh/dummy-ride-request: Expose to UI & City configurable dummy ride request

### PR DESCRIPTION
**Pull request by jaypalmudaliyar24**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added city specific dummy ride request data in `transporter_config` table.

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Motivation and Context

This PR makes the dummy data for ride request configurable based on driver's merchant_operating_city.

## How did you test it?

Tested locally.

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Location.hs [hunk 1]
```diff
@@ -59,9 +61,84 @@ data LocationAPIEntity = LocationAPIEntity
...
+      street = Just "6th Block",
+      area = Just "Koramangala, 6th Block",
+      areaCode = Just "560095",
+      city = Just "Bengaluru",
+      state = Just "Karnataka 560095",
+      country = Just "India",
+      fullAddress = Just "Rohit 17th F Main Rd, 6th Block, Koramangala, Bengaluru, Karnataka 560095, 560095, India"
+    }
+
+testValueForDummyLocationDbData :: String
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Location.hs:127]: Why do we need this?

File: Backend/dev/migrations/dynamic-offer-driver-app/0343-add-dummy-location-data-to-transporter-config.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- Add new column
+ALTER TABLE atlas_driver_offer_bpp.transporter_config
+ADD COLUMN dummy_from_location text;
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0343-add-dummy-location-data-to-transporter-config.sql:3]: Store it as JSON.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Beam/Merchant/TransporterConfig.hs [hunk 3]
```diff
@@ -135,7 +135,9 @@ data TransporterConfigT f = TransporterConfigT
     notificationRetryTimeGap :: B.C f Seconds,
     driverAutoPayExecutionTimeFallBack :: B.C f Seconds,
     orderAndNotificationStatusCheckFallBackTime :: B.C f Seconds,
-    kaptureDisposition :: B.C f Text
+    kaptureDisposition :: B.C f Text,
+    dummyFromLocation :: B.C f (Maybe Text),
```
[the remark this answers is not in the document]
Comment (hkmangla) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Beam/Merchant/TransporterConfig.hs:139]: Check for configJSON value inside MerchantServiceConfig.

File: Backend/cabal.project [hunk 4]
```diff
@@ -1,5 +1,4 @@
 packages:
-  app/alchemist
```
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/cabal.project:2]: While building, it was throwing error when it wasn't removed it.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Location.hs (hunk 1 above)
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Location.hs:127]: Kept it for checking parsing, will remove it.

File: Backend/cabal.project (hunk 4 above)
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/cabal.project]: Fixed.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Location.hs (hunk 1 above)
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Location.hs]: Fixed.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Beam/Merchant/TransporterConfig.hs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Beam/Merchant/TransporterConfig.hs]: Fixed.

**Review by hkmangla — DISMISSED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**